### PR TITLE
MMCA-4934 | Refactor and add unit test cases for DB patch service

### PR DESCRIPTION
**Pull request by Neozxc**
Task:
- Add unit test for DbPatchService
- Refactor DbPatchService
- Turn off **delete-guarantee-account-cache-documents** feature

### CODE REVIEW

**Comment by tyagikumarvineet** [comment]
Could you please turn off the feature in application.conf as we should need to turn off on local or it has not been updated on purpose?

File: test/services/DbPatchServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import play.api.test.Helpers.running
+import play.api.{Application, inject}
+import uk.gov.hmrc.mongo.MongoComponent
+import utils.Utils.emptyString
+
+class DbPatchServiceSpec extends SpecBase {
+
+  "Database Patch Service" should {
+
+    "delete documents when deleteGuaranteeAccountCacheDocuments is true" in new Setup {
```
Comment (tyagikumarvineet) [inline test/services/DbPatchServiceSpec.scala:34]: Better to update the test description to as below as feature flag info should be highlighted as that is the main bit here. Cause, deleteGuaranteeAccountCacheDocuments without any reference to config  is not adding that much value in the description. same for test starts at line 47

"delete documents when 'delete-guarantee-account-cache-documents' feature flag is true"

or "delete documents when config.deleteGuaranteeAccountCacheDocuments is true"

**Comment by tyagikumarvineet** [comment]
> Could you please turn off the feature in application.conf as we should need to turn off on local or it has not been updated on purpose?

Has been addressed.

File: test/services/DbPatchServiceSpec.scala (hunk 1 above)
Comment (Neozxc) [reply test/services/DbPatchServiceSpec.scala]: Renamed as per comment

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**